### ISSUE TITLE
Release for GHC 9.8.1

### ISSUE THREAD

**Issue by bgamari** [haskell/containers#971]
@treeowl, on reviewing GHC's submodules I noticed that the `containers` commit used by GHC 9.8.1 has a few substantive changes relative to the most recent release, 0.6.7. Unfortunately, this means that a Hackage revision will not be sufficient and we will instead need a proper release. Do you suppose this could happen sometime in the next week or so?

If a release is not possible then I can instead bump GHC back to 0.6.7 with a bounds bump revision (namely f61b0c9104a3c436361f56a0974c5eeef40c1b89).

Apologies for noticing this so late; my earlier audit from our discussion in #955 left me thinking that the only changes were in benchmarks and tests.

**Comment by treeowl** [comment]
Yes, we can make a release this week.

**Comment by treeowl** [comment]
I'll be able to make the release tomorrow.

**Comment by treeowl** [comment]
Released!

**Comment by Bodigrim** [comment]
@treeowl something seems wrong here: f5d0b13251291c3bd1ae396f3e6c8b0b9eaf58b0 says "Bump version to 0.7", but actually bumps to 0.6.8 only: 
https://github.com/haskell/containers/blob/f5d0b13251291c3bd1ae396f3e6c8b0b9eaf58b0/containers/containers.cabal#L1-L2

Hackage release seems correct however.

**Comment by treeowl** [comment]
Yes, that commit is broken. The next one fixes it, no?

**Comment by bgamari** [comment]
I very much wish I had noticed this mix-up this morning. I had assumed that the `master` commit was the final release; in hindsight I should have noticed that something was amiss when I noticed that there was no tag uploaded. @treeowl, if there is a next commit I'm afraid it wasn't pushed.

Unfortunately, the GHC release pipeline which I had planned to cut the release candidate from refers to this broken commit and will need to be scrapped. This  implies that we will need to push the release candidate by at least a day. However, I fear that the situation is even worse than this as none of the boot libraries' bounds current accommodate `containers-0.7`. Fixing this would very likely take weeks.

@treeowl, I don't believe a major release of `containers` is feasible at this juncture. At this point I think we should plan to do the bare minimum to get the release out the door (ideally just a Hackage revision). I believe the only thing preventing us from using 0.6.7 is the `deepseq < 1.5` constraint. Would you be willing to lift this in a revision?

**Comment by Bodigrim** [comment]
> However, I fear that the situation is even worse than this as none of the boot libraries' bounds current accommodate `containers-0.7`. Fixing this would very likely take weeks.

@bgamari  FWIW I think it's only `Cabal` who needs a bump for `containers`, and `Cabal-3.10.2.0` is pending a release anyway, might as well do it.

**Comment by Bodigrim** [comment]
I raised https://github.com/haskell/cabal/issues/9289 for `Cabal`, CC @Kleidukos.

**Comment by bgamari** [comment]
@Bodigrim beyond the problem of boot library bounds, there is also the matter of downstream users. Ideally we try not to bump major versions this late in the cycle since downstreams may have already started bumping their bounds to accommodate the new release.

**Comment by treeowl** [comment]
There is a change in `Data.Graph` that is at least theoretically breaking. I believe that was submitted by @Bodigrim . So either we can say that we are confident that won't actually break anything and call this minor, or we can back that out till next release. I don't remember the reason for the `deepseq` bump off the top of my head, and I can't look right now.

**Comment by bgamari** [comment]
@treeowl, the `deepseq` bounds bump is necessary because GHC 9.8 will ship with `deepseq-1.5`.

My proposal is that we simply create a `containers-0.6` branch, cherry-pick 8663795322622ac77cc78566185bffbc84e299f2 to it, and push a Hackage revision with this bump.

**Comment by Bodigrim** [comment]
> There is a change in `Data.Graph` that is at least theoretically breaking. I believe that was submitted by @Bodigrim . So either we can say that we are confident that won't actually break anything and call this minor, or we can back that out till next release.

I don't believe the change #953 can break anything in compile time, but I classified it as breaking back then because `instance {Show,Read} (SCC vertex)` is now different. E. g., golden tests comparing serialised `SCC` will get broken.

**Comment by treeowl** [comment]
@bgamari , most of the release is documentation and testing improvements, so I think just backing out the `SCC` change from 0.7 will produce a good 0.6. I'll have to look after my rehearsal.

**Comment by bgamari** [comment]
Thank you.

**Comment by treeowl** [comment]
@bgamari I backed out the potentially breaking commit on an 0.6.8 branch. I don't think any of the other changes could plausibly break anything.

**Comment by bgamari** [comment]
Will you be able to cut a release from this branch?
